jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -796,25 +796,41 @@ ecma_op_function_get_realm (const ecma_compiled_code_t *bytecode_header_p) /**<
 ecma_global_object_t *
 ecma_op_function_get_function_realm (ecma_object_t *func_obj_p) /**< function object */
 {
-  if (ecma_get_object_type (func_obj_p) == ECMA_OBJECT_TYPE_FUNCTION)
+  while (true)
   {
-    ecma_extended_object_t *ext_function_obj_p = (ecma_extended_object_t *) func_obj_p;
-    const ecma_compiled_code_t *bytecode_data_p = ecma_op_function_get_compiled_code (ext_function_obj_p);
-    return ecma_op_function_get_realm (bytecode_data_p);
-  }
+    if (ecma_get_object_type (func_obj_p) == ECMA_OBJECT_TYPE_FUNCTION)
+    {
+      ecma_extended_object_t *ext_function_obj_p = (ecma_extended_object_t *) func_obj_p;
+      const ecma_compiled_code_t *bytecode_data_p = ecma_op_function_get_compiled_code (ext_function_obj_p);
+      return ecma_op_function_get_realm (bytecode_data_p);
+    }
 
-  JERRY_ASSERT (ecma_get_object_type (func_obj_p) == ECMA_OBJECT_TYPE_NATIVE_FUNCTION);
+  #if ENABLED (JERRY_BUILTIN_PROXY)
+    if (ECMA_OBJECT_IS_PROXY (func_obj_p))
+    {
+      ecma_proxy_object_t *proxy_obj_p = (ecma_proxy_object_t *) func_obj_p;
+      if (ecma_is_value_null (proxy_obj_p->handler))
+      {
+        ecma_raise_type_error (ECMA_ERR_MSG ("Prototype from revoked Proxy is invalid."));
+        return NULL;
+      }
+      func_obj_p = ecma_get_object_from_value (proxy_obj_p->target);
+      continue;
+    }
+  #endif /* ENABLED (JERRY_BUILTIN_PROXY) */
+    JERRY_ASSERT (ecma_get_object_type (func_obj_p) == ECMA_OBJECT_TYPE_NATIVE_FUNCTION);
```

Comment:
This is better but please put an `if (ecma_get_object_type (func_obj_p) == ECMA_OBJECT_TYPE_NATIVE_FUNCTION)` check before `if (ECMA_OBJECT_IS_PROXY (func_obj_p))`. I think Proxy has the lowest chance.